clock/scheduledJobs/queueStatsActivityLogs.js

```diff
@@ -0,0 +1,260 @@
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * clock/scheduledJobs/queueStatsActivityLogs.js
+ *
+ * This file is for queue stats activity logs.
+ */
+
+const activityLogUtil = require('../../utils/activityLog');
+const queueTimeMillis95th = require('../../config').queueTimeMillis95th;
+const redis = require('../../cache/redisCache');
+const ZERO = 0;
+const ONE = 1;
+const TWO = 2;
+const FLOAT_TWO = 2.0;
+const PERCENTILE_95TH = 0.95;
+
+// Redis connection
+const client = redis.client.realtimeLogging;
+
+/**
+ * Convert array to string
+ *
+ * @param {array} array - array of integers
+ * @returns {string} string - string of array
+ */
+function arrayToString(array) {
+  return array.join(',');
+}
+
+/**
+ * Convert string to array
+ *
+ * @param {string} string - string of array
+ * @returns {array} array - array of integers
+ */
+function stringToArray(string) {
+  if (string.length) {
+    const array = string.split(',');
+    for (let i = 0; i < array.length; i++) {
+      if (!isNaN(array[i])) {
+        array[i] = Number(array[i]);
+      }
+    }
+
+    return array;
+  }
+
+  return [];
+}
+
+/**
+ * Convert string to Array then add element and again conver
+ * to string
+ *
+ * @param {string} string - string of array
+ * @param {integer} qt - queueTime
+ * @returns {string} string - string of array
+ */
+function addToArray(string, qt) {
+  const array = stringToArray(string);
+  array.push(qt);
+  return arrayToString(array);
+}
+
+/**
+ * Calculate timestamp i.e 2017.04.11.15.58.
+ *
+ * @returns {string} timestamp - timestamp string
+ */
+function createTimeStamp() {
+  const date = new Date();
+  return date.getUTCFullYear() + '.' + (date.getUTCMonth() + ONE) + '.' +
+    date.getUTCDate() + '.' + date.getUTCHours() + '.' + date.getUTCMinutes();
+}
+
+/**
+ * Calculate Statistics based on queue timing array.
+ *
+ * @param {array} qt - array of queue timings
+ * @returns {object} stats - statistics based on queue timing array
+ *     i.e median, average and 95th Percentile
+ */
+function calculateStats(qt) {
+  let median = 0;
+  let average = 0;
+  let percentile95th = 0;
+  const stats = {};
+  const length = qt.length;
+
+  // sort the queue time array
+  qt.sort((a, b) => a - b);
+
+  // calculate sum of queue time
+  const sum = qt.reduce((a, b) => a + b, ZERO);
+
+  // calculate middle of array
+  const middle = Math.floor(length / TWO);
+  average = sum / length;
+
+  // calculate median
+  if (length % TWO) {
+    median = qt[middle];
+  } else {
+    median = (qt[middle - ONE] + qt[middle]) / FLOAT_TWO;
+  }
+
+  // calculate 95th Percentile
+  const index = Math.round(PERCENTILE_95TH * length);
+  percentile95th = qt[index - ONE];
+
+  // construct stats object
+  stats.averageQueueTimeMillis = average.toFixed(TWO) + 'ms';
```

Comment:
we don't need "Millis" at the end of the attribute name AND the "ms" after the numbers
let's drop the "ms" after the numbers